re_data/command_line.py

```diff
@@ -299,21 +340,38 @@ def serve(port):
     default='',
     help="Extra markdown text to be added to the alert message"
 )
+@click.option(
+    '--re-data-target-dir',
+    type=click.STRING,
+    help="""
+        Which directory to store artefacts generated by re_data
+        Defaults to the 'target-path' used in dbt_project.yml
+    """
+)
 @add_options(dbt_flags)
-def slack(start_date, end_date, webhook_url, subtitle, **kwargs):
+def slack(start_date, end_date, webhook_url, subtitle, re_data_target_dir, **kwargs):
     start_date = str(start_date.date())
     end_date = str(end_date.date())
+
+    project_root = os.getcwd() if not kwargs.get('project_dir') else os.path.abspath(kwargs['project_dir'])
```

Comment:
it does seem to be quite repeated? Refactoring that to a function?

Reply:
this is done now.